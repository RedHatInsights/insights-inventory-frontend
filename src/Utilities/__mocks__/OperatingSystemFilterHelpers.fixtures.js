export const availableVersions = [
  {
    groupLabel: 'RHEL 6',
    label: 'RHEL 6.9',
    osName: 'RHEL',
    value: '6.9',
  },
  {
    groupLabel: 'RHEL 6',
    label: 'RHEL 6.10',
    osName: 'RHEL',
    value: '6.10',
  },
  {
    groupLabel: 'RHEL 7',
    label: 'RHEL 7.0',
    osName: 'RHEL',
    value: '7.0',
  },
  {
    groupLabel: 'RHEL 8',
    label: 'RHEL 8.5',
    osName: 'RHEL',
    value: '8.5',
  },
  {
    groupLabel: 'RHEL 8',
    label: 'RHEL 8.10',
    osName: 'RHEL',
    value: '8.10',
  },
  {
    groupLabel: 'CentOS Linux 7',
    label: 'CentOS Linux 7.9',
    osName: 'CentOS Linux',
    value: '7.9',
  },
  {
    groupLabel: 'CentOS Linux 8',
    label: 'CentOS Linux 8.6',
    osName: 'CentOS Linux',
    value: '8.6',
  },
];

export const testValue1 = {
  'RHEL 7': {
    'RHEL 7': true,
    '7.0': true,
    7.1: true,
    7.2: true,
  },
  'RHEL 8': {
    'RHEL 8': true,
    8.5: true,
    '8.10': true,
  },
  'RHEL 9': {
    'RHEL 9': true,
    9.1: true,
  },
  'CentOS Linux 7': {
    'CentOS Linux 7': true,
    7.6: true,
    7.9: true,
  },
  'CentOS Linux 8': {
    'CentOS Linux 8': true,
    8.3: true,
    8.9: true,
  },
};

/*eslint-disable prettier/prettier*/
export const testValue2 = {
  'RHEL 6': {
    'RHEL 6': false,
    '6.9': true,
    '6.10': false,
  },
  'RHEL 7': {
    'RHEL 7': true,
    '7.0': true,
  },
  'RHEL 8': {
    'RHEL 8': true,
    '8.5': true,
    '8.10': true,
  },
  'CentOS Linux 7': {
    'CentOS Linux 7': true,
    '7.9': true,
  },
  'CentOS Linux 8': {
    'CentOS Linux 8': false,
    '8.6': false,
  },
};

export const testValue3 = {
  'RHEL 6': {
    'RHEL 6': false,
    '6.9': false,
    '6.10': false,
  },
  'RHEL 7': {
    'RHEL 7': false,
    '7.0': false,
  },
  'RHEL 8': {
    'RHEL 8': false,
    '8.5': false,
    '8.10': false,
  },
  'CentOS Linux 7': {
    'CentOS Linux 7': false,
    '7.9': false,
  },
  'CentOS Linux 8': {
    'CentOS Linux 8': false,
    '8.6': false,
  },
};

export const testValue4 = {
  'RHEL 6': {
    'RHEL 6': false,
    '6.9': false,
    '6.10': false,
  },
  'RHEL 7': {
    'RHEL 7': false,
    '7.0': false,
  },
  'RHEL 8': {
    'RHEL 8': true,
    '8.5': true,
    '8.10': true,
  },
  'CentOS Linux 7': {
    'CentOS Linux 7': false,
    '7.9': false,
  },
  'CentOS Linux 8': {
    'CentOS Linux 8': false,
    '8.6': false,
  },
};
/*eslint-enable prettier/prettier*/

export const groupSelection1 = [
  { osName: 'RHEL', osGroup: 'RHEL 7', value: '7.0' },
  { osName: 'RHEL', osGroup: 'RHEL 7', value: '7.1' },
  { osName: 'RHEL', osGroup: 'RHEL 7', value: '7.2' },
  { osName: 'RHEL', osGroup: 'RHEL 8', value: '8.5' },
  { osName: 'RHEL', osGroup: 'RHEL 8', value: '8.10' },
  { osName: 'RHEL', osGroup: 'RHEL 9', value: '9.1' },
  { osName: 'CentOS Linux', osGroup: 'CentOS Linux 7', value: '7.6' },
  { osName: 'CentOS Linux', osGroup: 'CentOS Linux 7', value: '7.9' },
  { osName: 'CentOS Linux', osGroup: 'CentOS Linux 8', value: '8.3' },
  { osName: 'CentOS Linux', osGroup: 'CentOS Linux 8', value: '8.9' },
];

export const groupSelection2 = [
  { osName: 'RHEL', osGroup: 'RHEL 6', value: '6.9' },
  { osName: 'RHEL', osGroup: 'RHEL 7', value: '7.0' },
  { osName: 'RHEL', osGroup: 'RHEL 8', value: '8.5' },
  { osName: 'RHEL', osGroup: 'RHEL 8', value: '8.10' },
  { osName: 'CentOS Linux', osGroup: 'CentOS Linux 7', value: '7.9' },
];
