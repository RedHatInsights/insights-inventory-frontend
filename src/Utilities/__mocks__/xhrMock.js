const createXhrMock = () => {
  const xhrMockObj = {
    open: jest.fn(),
    send: jest.fn(),
    setRequestHeader: jest.fn(),
    readyState: 4,
    status,
    response: {},
  };

  return () => xhrMockObj;
};

export default createXhrMock;
