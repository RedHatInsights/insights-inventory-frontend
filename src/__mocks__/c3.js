// eslint-disable-next-line no-undef
module.exports = () => 'c3';

