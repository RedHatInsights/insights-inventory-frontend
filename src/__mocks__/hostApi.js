import { hosts, tags } from '../api/api';
import MockAdapter from 'axios-mock-adapter';

export const mock = new MockAdapter(hosts.axios, { onNoMatch: 'throwException' });
export const mockTags = new MockAdapter(tags.axios, { onNoMatch: 'throwException' });
