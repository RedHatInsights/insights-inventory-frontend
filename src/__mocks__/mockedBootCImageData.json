[
    {
        "image": "India Pale Ale",
        "hashCommitCount": 2,
        "systemCount": 3,
        "hashes": [
            {
                "image_digest": "Hazy",
                "hashSystemCount": 2
            },
            {
                "image_digest": "West Coast",
                "hashSystemCount": 1
            }
        ]
    },
    {
        "image": "Belgian",
        "hashCommitCount": 1,
        "systemCount": 1,
        "hashes": [
            {
                "image_digest": "Dubbel",
                "hashSystemCount": 1
            }
        ]
    },
    {
        "image": "Stout",
        "hashCommitCount": 1,
        "systemCount": 1,
        "hashes": [
            {
                "image_digest": "Bourbon Barrell Aged",
                "hashSystemCount": 1
            }
        ]
    }
]