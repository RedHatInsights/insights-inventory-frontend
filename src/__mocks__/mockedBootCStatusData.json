{
    "total": 5,
    "count": 5,
    "page": 1,
    "per_page": 50,
    "results": [
        {
            "id": "6de30d78-f052-41d4-b123-125017443a98",
            "system_profile": {
                "arch": "x86_64",
                "cpu_flags": [ "fpu" ],
                "yum_repos": [
                    {
                        "name": "Extra Packages for Enterprise Linux 7 - Testing - $basearch",
                        "enabled": false,
                        "gpgcheck": true
                    }
                ],
                "os_release": "7.4",
                "bios_vendor": "SeaBIOS",
                "bios_version": "1.10.2-2.fc27",
                "kernel_modules": [ "veth" ],
                "last_boot_time": "2018-03-14T17:06:10",
                "number_of_cpus": 1,
                "cores_per_socket": 1,
                "enabled_services": [ "auditd" ],
                "bios_release_date": "04\/01\/2014",
                "number_of_sockets": 1,
                "os_kernel_version": "3.10.0",
                "running_processes": [ "qmgr" ],
                "satellite_managed": false,
                "installed_packages": [ "0:GeoIP-1.5.0-11.el7" ],
                "installed_services": [ "arp-ethers" ],
                "network_interfaces": [
                    {
                        "mtu": 65536,
                        "name": "lo",
                        "type": "loopback",
                        "state": "UNKNOWN",
                        "mac_address": "00:00:00:00:00:00",
                        "ipv4_addresses": [
                            "127.0.0.1"
                        ],
                        "ipv6_addresses": [
                            "::1"
                        ]
                    }
                ],
                "infrastructure_type": "virtual",
                "system_memory_bytes": 512073728,
                "infrastructure_vendor": "qemu",
                "system_purpose": {
                    "usage": "Production"
                },
                "bootc_status": {
                    "booted": {
                        "image": "India Pale Ale",
                        "image_digest": "Hazy"
                    }
                }
            }
        },
        {
            "id": "08a06e04-afbb-4e3c-b923-91aa1c17acc1",
            "system_profile": {
                "arch": "x86_64",
                "cpu_flags": [ "fpu" ],
                "yum_repos": [
                    {
                        "name": "Extra Packages for Enterprise Linux 7 - Testing - $basearch",
                        "enabled": false,
                        "gpgcheck": true
                    }
                ],
                "os_release": "8.4",
                "bios_vendor": "SeaBIOS",
                "bios_version": "1.10.2-2.fc27",
                "kernel_modules": [ "veth" ],
                "last_boot_time": "2018-03-14T17:06:10",
                "number_of_cpus": 1,
                "cores_per_socket": 1,
                "enabled_services": [ "auditd" ],
                "bios_release_date": "04\/01\/2014",
                "number_of_sockets": 1,
                "os_kernel_version": "3.10.0",
                "running_processes": [ "qmgr" ],
                "satellite_managed": false,
                "installed_packages": [ "0:GeoIP-1.5.0-11.el7" ],
                "installed_services": [ "arp-ethers" ],
                "network_interfaces": [
                    {
                        "mtu": 65536,
                        "name": "lo",
                        "type": "loopback",
                        "state": "UNKNOWN",
                        "mac_address": "00:00:00:00:00:00",
                        "ipv4_addresses": [
                            "127.0.0.1"
                        ],
                        "ipv6_addresses": [
                            "::1"
                        ]
                    }
                ],
                "infrastructure_type": "virtual",
                "system_memory_bytes": 512073728,
                "infrastructure_vendor": "qemu",
                "system_purpose": {
                    "usage": "Production"
                },
                "bootc_status": {
                    "booted": {
                        "image": "India Pale Ale",
                        "image_digest": "Hazy"
                    }
                }
            }
        },
        {
            "id": "4ef68d04-79fa-42ea-a0fe-b40ee257ff0e",
            "system_profile": {
                "arch": "x86_64",
                "cpu_flags": [ "fpu" ],
                "yum_repos": [
                    {
                        "name": "Extra Packages for Enterprise Linux 7 - Testing - $basearch",
                        "enabled": false,
                        "gpgcheck": true
                    }
                ],
                "os_release": "8.5",
                "bios_vendor": "SeaBIOS",
                "bios_version": "1.10.2-2.fc27",
                "kernel_modules": [ "veth" ],
                "last_boot_time": "2018-03-14T17:06:10",
                "number_of_cpus": 1,
                "cores_per_socket": 1,
                "enabled_services": [ "auditd" ],
                "bios_release_date": "04\/01\/2014",
                "number_of_sockets": 1,
                "os_kernel_version": "3.10.0",
                "running_processes": [ "qmgr" ],
                "satellite_managed": false,
                "installed_packages": [ "0:GeoIP-1.5.0-11.el7" ],
                "installed_services": [ "arp-ethers" ],
                "network_interfaces": [
                    {
                        "mtu": 65536,
                        "name": "lo",
                        "type": "loopback",
                        "state": "UNKNOWN",
                        "mac_address": "00:00:00:00:00:00",
                        "ipv4_addresses": [
                            "127.0.0.1"
                        ],
                        "ipv6_addresses": [
                            "::1"
                        ]
                    }
                ],
                "infrastructure_type": "virtual",
                "system_memory_bytes": 512073728,
                "infrastructure_vendor": "qemu",
                "system_purpose": {
                    "usage": "Production"
                },
                "bootc_status": {
                    "booted": {
                        "image": "India Pale Ale",
                        "image_digest": "West Coast"
                    }
                }
            }
        },
        {
            "id": "b6197999-06f8-4301-a936-651832aa8b31",
            "system_profile": {
                "arch": "x86_64",
                "cpu_flags": [ "fpu" ],
                "yum_repos": [
                    {
                        "name": "Extra Packages for Enterprise Linux 7 - Testing - $basearch",
                        "enabled": false,
                        "gpgcheck": true
                    }
                ],
                "os_release": "8.5",
                "bios_vendor": "SeaBIOS",
                "bios_version": "1.10.2-2.fc27",
                "kernel_modules": [ "veth" ],
                "last_boot_time": "2018-03-14T17:06:10",
                "number_of_cpus": 1,
                "cores_per_socket": 1,
                "enabled_services": [ "auditd" ],
                "bios_release_date": "04\/01\/2014",
                "number_of_sockets": 1,
                "os_kernel_version": "3.10.0",
                "running_processes": [ "qmgr" ],
                "satellite_managed": false,
                "installed_packages": [ "0:GeoIP-1.5.0-11.el7" ],
                "installed_services": [ "arp-ethers" ],
                "network_interfaces": [
                    {
                        "mtu": 65536,
                        "name": "lo",
                        "type": "loopback",
                        "state": "UNKNOWN",
                        "mac_address": "00:00:00:00:00:00",
                        "ipv4_addresses": [
                            "127.0.0.1"
                        ],
                        "ipv6_addresses": [
                            "::1"
                        ]
                    }
                ],
                "infrastructure_type": "virtual",
                "system_memory_bytes": 512073728,
                "infrastructure_vendor": "qemu",
                "system_purpose": {
                    "usage": "Production"
                },
                "bootc_status": {
                    "booted": {
                        "image": "Belgian",
                        "image_digest": "Dubbel"
                    }
                }
            }
        },
        {
            "id": "8499b198-5352-4eda-8650-ca09e002fdc1",
            "system_profile": {
                "arch": "x86_64",
                "cpu_flags": [ "fpu" ],
                "yum_repos": [
                    {
                        "name": "Extra Packages for Enterprise Linux 7 - Testing - $basearch",
                        "enabled": false,
                        "gpgcheck": true
                    }
                ],
                "os_release": "8.2",
                "bios_vendor": "SeaBIOS",
                "bios_version": "1.10.2-2.fc27",
                "kernel_modules": [ "veth" ],
                "last_boot_time": "2018-03-14T17:06:10",
                "number_of_cpus": 1,
                "cores_per_socket": 1,
                "enabled_services": [ "auditd" ],
                "bios_release_date": "04\/01\/2014",
                "number_of_sockets": 1,
                "os_kernel_version": "3.10.0",
                "running_processes": [ "qmgr" ],
                "satellite_managed": false,
                "installed_packages": [ "0:GeoIP-1.5.0-11.el7" ],
                "installed_services": [ "arp-ethers" ],
                "network_interfaces": [
                    {
                        "mtu": 65536,
                        "name": "lo",
                        "type": "loopback",
                        "state": "UNKNOWN",
                        "mac_address": "00:00:00:00:00:00",
                        "ipv4_addresses": [
                            "127.0.0.1"
                        ],
                        "ipv6_addresses": [
                            "::1"
                        ]
                    }
                ],
                "infrastructure_type": "virtual",
                "system_memory_bytes": 512073728,
                "infrastructure_vendor": "qemu",
                "system_purpose": {
                    "usage": "Production"
                },
                "bootc_status": {
                    "booted": {
                        "image": "Stout",
                        "image_digest": "Bourbon Barrell Aged"
                    }
                }
            }
        }
    ]
}
