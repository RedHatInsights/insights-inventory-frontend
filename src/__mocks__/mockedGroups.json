{
  "total": 1,
  "count": 1,
  "page": 1,
  "per_page": 50,
  "results": [
    {
      "name": "aute",
      "id": "FC7CDc2A-EbCd-36F5-f2cc-72f7f2fB960F",
      "host_ids": [
        "2Ab5EF47-a25f-eC8D-459D-e48Ca051b1F2",
        "EdE9F657-B7FD-E56c-2E66-6a93ee8aabcC"
      ],
      "created_at": "2009-01-31T23:00:00.0Z",
      "updated_at": "1981-07-06T22:00:00.0Z"
    }
  ]
}
