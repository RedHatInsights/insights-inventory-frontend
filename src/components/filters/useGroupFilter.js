/* eslint-disable camelcase */
import union from 'lodash/union';
import { useEffect, useMemo, useState } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { fetchGroups } from '../../store/inventory-actions';
import { HOST_GROUP_CHIP } from '../../Utilities/index';

//for attaching this filter to the redux
export const groupFilterState = { groupHostFilter: null };
export const GROUP_FILTER = 'GROUP_FILTER';
export const groupFilterReducer = (_state, { type, payload }) => ({
    ...type === GROUP_FILTER && {
        groupHostFilter: payload
    }
});

//receive the array of selected groups and return chips based on the name of selected groups
export const buildHostGroupChips = (selectedGroups = []) => {
    //we use new Set to make sure that chips are unique
    const chips = [...selectedGroups]?.map((group) => ({ name: group, value: group }));
    return chips?.length > 0
        ? [
            {
                category: 'Group',
                type: HOST_GROUP_CHIP,
                chips
            }
        ]
        : [];
};

const useGroupFilter = (apiParams = []) => {
    const dispatch = useDispatch();
    useEffect(() => {
        dispatch(fetchGroups(apiParams));
    }, []);
    //fetched values
    const fetchedValues = useSelector(({ groups })  => groups?.data?.results);
    //selected are the groups we selected
    const [selected, setSelected] = useState([]);
    //buildHostGroupsValues build an array of objects to populate dropdown
    const [buildHostGroupsValues, setBuildHostGroupsValues] = useState([]);
    //hostGroupValue is used for config items
    useEffect(() => {
        setBuildHostGroupsValues((fetchedValues || []).reduce((acc, group) => {
            acc.push({ label: group.name, value: group.name });
            return acc;
        }, []));
    }, [fetchedValues, selected]);
    //this is used in the filter config as a way to select values onChange
    const onHostGroupsChange = (event, selection, item) => {
        setSelected(union(selection, item));
    };

    const chips = useMemo(() => buildHostGroupChips(selected), [selected]);
    //chips that is built for the filter config

    //hostGroupConfig is a config that we use in EntityTableToolbar.js
    const hostGroupConfig = useMemo(() => ({
        label: 'Group',
        value: 'group-host-filter',
        type: 'checkbox',
        filterValues: {
            onChange: (event, value, item) => {
                onHostGroupsChange(event, value, item);
            },
            value: selected,
            items: buildHostGroupsValues
        }
    }), [selected, buildHostGroupsValues]);

    //setSelectedValues is used for selecting and deleting values
    const setSelectedValues = (currentValue = []) => {
        setSelected(currentValue);
    };

    return [hostGroupConfig, chips, selected, setSelectedValues];
};

export default useGroupFilter;
