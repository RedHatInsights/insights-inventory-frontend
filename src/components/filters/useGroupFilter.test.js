/* eslint-disable camelcase */
import { act, renderHook } from '@testing-library/react-hooks';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import { createPromise as promiseMiddleware } from 'redux-promise-middleware';
import { mockSystemProfile } from '../../__mocks__/hostApi';
import useGroupFilter from './useGroupFilter';

describe('useGroupFilter', () => {
    const mockStore = configureStore([promiseMiddleware()]);
    beforeEach(() => {
        mockSystemProfile.onGet().replyOnce(200);
    });

    describe('with groups yet not loaded', () => {
        const wrapper = ({ children }) => (
            <Provider store={mockStore({})}>
                {children}
            </Provider>
        );

        it('should return empty state value', () => {
            const { result } = renderHook(useGroupFilter, { wrapper });
            expect(result.current).toMatchSnapshot();
        });
    });

    describe('with groups loaded', () => {
        const wrapper = ({ children }) => (
            <Provider
                store={mockStore({
                    groups: {
                        data: {
                            page: 1,
                            count: 50,
                            results: [{
                                created_at: '2019-02-18T23:00:00.0Z',
                                id: '4f5B88dBe1D4eB732B388abc1Baa4BAc',
                                updated_at: '1960-11-06T23:00:00.0Z',
                                org_id: 'nostrud in deserunt',
                                account: 'ullamco dolore pariatur sint',
                                host_ids: [
                                    '5f20569C-9E96-C794-887c-1Cb5D6e220b2',
                                    'D34e68beFc2d4fF1C19CF4CaF913d1b6',
                                    '76D3AcBB-5B4C-8BB8-f3BB-69808CcFa84A',
                                    '185dC76d-D6B8-059b-1bCb-685D2edd0CEa',
                                    '50E0cAaDCace55e07AebC7Bcc40cfdAD',
                                    'be8e0A09AdEE9a23fe65972F43D31bDA',
                                    '8B6764001Bf6d0bA4E6aEdCAf207bf71',
                                    '80CffFbd-04C9-cBB6-5Bf5-Dc0eBbDBba70',
                                    '55d8323C-dEbC-B10e-f88e-23ba37068C9f',
                                    'C5A8Fd55a0B6fa3A3a748cb0a4cc20BB',
                                    '1dF767C34BF222E65BD46BcC3A62de4b',
                                    '4d44747b1D8D1648EC5713e983984a6C',
                                    'b89A20e9da89Eb9862502748ca9aa0bf',
                                    'F24dc6A1548B40dF453FdBEa4DD5CbEB',
                                    '2Ba0BCE6-F8cC-FAFE-Ed72-f7BF01dC60A4',
                                    '1f255AB7-d49f-f8Ff-48eD-B9c6E4674Fe2',
                                    '8277DD2f3Fc0aaE89D601400E86E4E86',
                                    '3Be6b8FE-8B32-240d-9C0f-0c074baC5EBb',
                                    '94c4bBCB31D0dfa37e955Dc94eAd3Dc0',
                                    '2B0725BE87671E4DffE19F0f7fc8aFe7'
                                ],
                                name: 'nisi ut consequat ad'
                            },
                            {
                                created_at: '2019-02-18T23:00:00.0Z',
                                id: '4f5B88dBe1D4eB732B388abc1Baa4BAc',
                                updated_at: '1960-11-06T23:00:00.0Z',
                                org_id: 'nostrud in deserunt',
                                account: 'ullamco dolore pariatur sint',
                                host_ids: [
                                    '5f20569C-9E96-C794-887c-1Cb5D6e220b2',
                                    'D34e68beFc2d4fF1C19CF4CaF913d1b6',
                                    '76D3AcBB-5B4C-8BB8-f3BB-69808CcFa84A',
                                    '185dC76d-D6B8-059b-1bCb-685D2edd0CEa',
                                    '50E0cAaDCace55e07AebC7Bcc40cfdAD',
                                    'be8e0A09AdEE9a23fe65972F43D31bDA',
                                    '8B6764001Bf6d0bA4E6aEdCAf207bf71',
                                    '80CffFbd-04C9-cBB6-5Bf5-Dc0eBbDBba70',
                                    '55d8323C-dEbC-B10e-f88e-23ba37068C9f',
                                    'C5A8Fd55a0B6fa3A3a748cb0a4cc20BB',
                                    '1dF767C34BF222E65BD46BcC3A62de4b',
                                    '4d44747b1D8D1648EC5713e983984a6C',
                                    'b89A20e9da89Eb9862502748ca9aa0bf',
                                    'F24dc6A1548B40dF453FdBEa4DD5CbEB',
                                    '2Ba0BCE6-F8cC-FAFE-Ed72-f7BF01dC60A4',
                                    '1f255AB7-d49f-f8Ff-48eD-B9c6E4674Fe2',
                                    '8277DD2f3Fc0aaE89D601400E86E4E86',
                                    '3Be6b8FE-8B32-240d-9C0f-0c074baC5EBb',
                                    '94c4bBCB31D0dfa37e955Dc94eAd3Dc0',
                                    '2B0725BE87671E4DffE19F0f7fc8aFe7'
                                ],
                                name: 'nisi ut consequat ad1'
                            },
                            {
                                created_at: '2019-02-18T23:00:00.0Z',
                                id: '4f5B88dBe1D4eB732B388abc1Baa4BAc',
                                updated_at: '1960-11-06T23:00:00.0Z',
                                org_id: 'nostrud in deserunt',
                                account: 'ullamco dolore pariatur sint',
                                host_ids: [
                                    '5f20569C-9E96-C794-887c-1Cb5D6e220b2',
                                    'D34e68beFc2d4fF1C19CF4CaF913d1b6',
                                    '76D3AcBB-5B4C-8BB8-f3BB-69808CcFa84A',
                                    '185dC76d-D6B8-059b-1bCb-685D2edd0CEa',
                                    '50E0cAaDCace55e07AebC7Bcc40cfdAD',
                                    'be8e0A09AdEE9a23fe65972F43D31bDA',
                                    '8B6764001Bf6d0bA4E6aEdCAf207bf71',
                                    '80CffFbd-04C9-cBB6-5Bf5-Dc0eBbDBba70',
                                    '55d8323C-dEbC-B10e-f88e-23ba37068C9f',
                                    'C5A8Fd55a0B6fa3A3a748cb0a4cc20BB',
                                    '1dF767C34BF222E65BD46BcC3A62de4b',
                                    '4d44747b1D8D1648EC5713e983984a6C',
                                    'b89A20e9da89Eb9862502748ca9aa0bf',
                                    'F24dc6A1548B40dF453FdBEa4DD5CbEB',
                                    '2Ba0BCE6-F8cC-FAFE-Ed72-f7BF01dC60A4',
                                    '1f255AB7-d49f-f8Ff-48eD-B9c6E4674Fe2',
                                    '8277DD2f3Fc0aaE89D601400E86E4E86',
                                    '3Be6b8FE-8B32-240d-9C0f-0c074baC5EBb',
                                    '94c4bBCB31D0dfa37e955Dc94eAd3Dc0',
                                    '2B0725BE87671E4DffE19F0f7fc8aFe7'
                                ],
                                name: 'nisi ut consequat ad2'
                            }
                            ],
                            total: 100
                        }
                    }
                })}
            >
                {children}
            </Provider>
        );

        it('should match snapshot', () => {
            const { result } = renderHook(useGroupFilter, { wrapper });
            expect(result.current).toMatchSnapshot();
        });

        it('should return correct chips array, current value and value setter', () => {
            const { result } = renderHook(useGroupFilter, { wrapper });
            const [, chips, value, setValue] = result.current;
            expect(chips.length).toBe(0);
            expect(value.length).toBe(0);
            act(() => {
                setValue(['nisi ut consequat ad1']);
            });
            const [, chipsUpdated, valueUpdated] = result.current;
            expect(chipsUpdated.length).toBe(1);
            expect(valueUpdated).toEqual(['nisi ut consequat ad1']);
            expect(chipsUpdated).toMatchSnapshot();
        });

        it('should return empty state value if no groups obtained', () => {
            const wrapper = ({ children }) => (
                <Provider
                    store={mockStore({
                        groups: {
                            data: {
                                page: 1,
                                count: 50,
                                results: [],
                                total: 100
                            }
                        }
                    })}
                >
                    {children}
                </Provider>
            );
            const { result } = renderHook(useGroupFilter, { wrapper });
            expect(result.current).toMatchSnapshot();
        });
    });
});
