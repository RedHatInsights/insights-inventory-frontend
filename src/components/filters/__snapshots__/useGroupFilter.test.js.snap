// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`useGroupFilter with groups loaded should match snapshot 1`] = `
Array [
  Object {
    "filterValues": Object {
      "items": Array [
        Object {
          "label": "nisi ut consequat ad",
          "value": "nisi ut consequat ad",
        },
        Object {
          "label": "nisi ut consequat ad1",
          "value": "nisi ut consequat ad1",
        },
        Object {
          "label": "nisi ut consequat ad2",
          "value": "nisi ut consequat ad2",
        },
      ],
      "onChange": [Function],
      "value": Array [],
    },
    "label": "Group",
    "type": "checkbox",
    "value": "group-host-filter",
  },
  Array [],
  Array [],
  [Function],
]
`;

exports[`useGroupFilter with groups loaded should return correct chips array, current value and value setter 1`] = `
Array [
  Object {
    "category": "Group",
    "chips": Array [
      Object {
        "name": "nisi ut consequat ad1",
        "value": "nisi ut consequat ad1",
      },
    ],
    "type": "host_group",
  },
]
`;

exports[`useGroupFilter with groups loaded should return empty state value if no groups obtained 1`] = `
Array [
  Object {
    "filterValues": Object {
      "items": Array [],
      "onChange": [Function],
      "value": Array [],
    },
    "label": "Group",
    "type": "checkbox",
    "value": "group-host-filter",
  },
  Array [],
  Array [],
  [Function],
]
`;

exports[`useGroupFilter with groups yet not loaded should return empty state value 1`] = `
Array [
  Object {
    "filterValues": Object {
      "items": Array [],
      "onChange": [Function],
      "value": Array [],
    },
    "label": "Group",
    "type": "checkbox",
    "value": "group-host-filter",
  },
  Array [],
  Array [],
  [Function],
]
`;
